---
title: "Sex and Age but not Body Mass Index (BMI) are Significant Predictors of Serum Retinol Binding Protein 4 (RBP4) and Transthyretin (TTR) Concentrations"
subtitle: "Statistical Analysis Code and Output"
author: "Aprajita S. Yadav"
date: "`r Sys.Date()`"
output:
  html_document:
    toc: true
    toc_depth: 3
    df_print: kable
editor_options: 
  markdown: 
    wrap: 72
---

\newpage

<!-- For PDF -->

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo=TRUE)
knitr::opts_chunk$set(warning = FALSE, message = FALSE)

##load libraries
library(dplyr)
library(car)
library(forcats)
library(tidyverse)
library(ggplot2)
library(ggpubr)
library(ggeasy)
library(rcompanion)
library(rigr)
library(rgl)
library(broom)
library(flextable)
library(knitr)
library(kableExtra)
library(Hmisc)
library(GGally)
library(grid)
library(officer)
```

# Load Data

```{r load data, include=FALSE}
# load data and make sure variables are read appropriately
###load data
vao <- read.csv ("C:/Users/isolab/UW/isolab - Aprajita Yadav/VAO/Retinoids/R data_and_analysis/Data_for_R.csv", header = TRUE)
names (vao)

###Sex, weight loss, snps as factors
vao$Sex<-as.factor(vao$Sex)
fct_relevel(vao$Sex,c("Female","Male"))
str(vao$Sex)

vao$Wt_change
vao$Wt_change<-as.factor(vao$Wt_change)
str(vao$Wt_change)

vao$RBP4snp<-as.factor(vao$RBP4.snp)
str(vao$RBP4snp)

vao$TTRsnp<-as.factor(vao$TTR.snp)
str(vao$TTRsnp)

vao$PNPLAsnp<-as.factor(vao$PNPLA3.snp)
str(vao$PNPLAsnp)

vao$RBP4.snp<-as.numeric(vao$RBP4snp)
vao$TTR.snp<-as.numeric(vao$TTRsnp)
vao$PNPLA3.snp<-as.numeric(vao$PNPLAsnp)

vao$RBP4_TTR<-vao$RBP4.TTR
vao$hRBP4<-vao$ROL.RBP4
vao$hRBP4_TTR<-vao$ROL.RBP4.TTR
vao$TTR_u4<-vao$TTR_u * 4 #unbound TTR not assumed to be in tetramer

#log data sets 
vao$LHOMAIR<-log10(vao$HOMAIR)
vao$LhRBP4<-log10(vao$hRBP4)
vao$LRBP4_TTR<-log10(vao$RBP4_TTR)
vao$LhRBP4_TTR<-log10(vao$hRBP4_TTR)
vao$LRBP4u<-log10(vao$RBP4_u)
vao$LROLu<-log10(vao$ROL_u)
vao$LTTRu<-log10(vao$TTR_u)

###convert Age, ast, alt to numeric
vao$Age<-as.numeric(vao$Age)
vao$AST<-as.numeric(vao$AST)
vao$ALT<-as.numeric(vao$ALT)
vao$NAFLDFS<-as.numeric(vao$NAFLDFS)
vao$HOMAIR<-as.numeric(vao$HOMAIR)
```

## Determine Normality

for Shapiro-Wilk test p-value\<0.05 indicates variables does not have normal distribution

```{r determine normality, echo=FALSE}
##determine normality, for Shapiro-Wilk test p-value>0.05 data is normally distributed
sq<-function(p) {p+theme(aspect.ratio=1)} #to make plots square

qqplot_array<-ggarrange(
  sq(ggqqplot(vao$BMI)), sq(ggqqplot(vao$HOMAIR)), sq(ggqqplot(vao$Age)), 
  sq(ggqqplot(vao$eGFR)),sq(ggqqplot(vao$Adiponectin)), sq(ggqqplot(vao$Leptin)), 
  sq(ggqqplot(vao$NAFLDFS)),
  ncol=4, nrow=2, 
  labels = c("BMI", "HOMA-IR", "Age", "eGFR", "Adiponectin", 
             "Leptin", "NAFLD-FS"),
  label.x=0, #center labels
  label.y=1) #labels above plots
qqplot_array<-annotate_figure(qqplot_array, top=text_grob(
  "Q-Q Plots for Variables", face="bold", size=10),
  fig.lab="",
  fig.lab.face="plain") + theme(plot.margin=unit(c(1,1,1,1),"mm"))
qqplot_array

normalitysummary<-bind_rows(
  mutate(tidy(shapiro.test(vao$BMI)),data="BMI"),
  mutate(tidy(shapiro.test(vao$HOMAIR)),data="HOMA-IR"),
  mutate(tidy(shapiro.test(vao$Age)),data="Age"),
  mutate(tidy(shapiro.test(vao$eGFR)),data="eGFR"),
  mutate(tidy(shapiro.test(vao$Adiponectin)),data="Adiponectin"),
  mutate(tidy(shapiro.test(vao$Leptin)),data="Leptin"),
  mutate(tidy(shapiro.test(vao$NAFLDFS)),data="NAFLD-FS"),
  )%>%
 mutate(p.value=ifelse(p.value<0.001, formatC(p.value, format = "e", 
                                               digits = 2), round(p.value, 3)))

normality_table <- normalitysummary %>%
  select(data, p.value) %>%
  flextable() %>%
  set_header_labels(data = "Variable", p.value = "p-value") %>%
  add_header_lines("Shapiro-Wilk Normality Test") %>%
  add_footer_lines("Note: data with p-value<0.05 will be log-transformed") %>%
  fontsize(part="footer", size=9)%>% align(part="header", align="center")%>%
  set_table_properties(layout = "autofit")

#display the combined results
normality_table

#log data sets that were not normal
vao$LHOMAIR<-log10(vao$HOMAIR)
vao$Adiponectin<-log10(vao$Adiponectin)
vao$Leptin<-log10(vao$Leptin)
vao<-vao %>% 
  mutate(across(c(LHOMAIR,Adiponectin,Leptin,NAFLDFS), as.numeric))
```

\newpage

<!-- For PDF -->

## Determine Variable Correlation

```{r determine if variables correlate with each other, echo=FALSE}
##anova to determine if Sex is correlated with other variables
#HOMAIR, adiponectin, leptin, AST and ALT are log-transformed because they do not follow a normal distribution
AOV_Sex<-bind_rows(
  mutate(tidy(aov(BMI ~ Sex, vao)),data="BMI"),
  mutate(tidy(aov(LHOMAIR~Sex,vao)),data="HOMA-IR"),
  mutate(tidy(aov(Age ~ Sex, vao)),data="Age"),
  mutate(tidy(aov(eGFR~Sex,vao)),data="eGFR"),
  mutate(tidy(aov(Adiponectin~Sex,vao)),data="Adiponectin"),
  mutate(tidy(aov(Leptin~Sex,vao)),data="Leptin"),
  mutate(tidy(aov(NAFLDFS~Sex,vao)),data="NAFLD-FS"))%>%
  filter(term !="Residuals") %>% select(data,p.value) %>%
  mutate(p.value=ifelse(p.value<0.01, formatC(p.value, format = "e", digits = 2), round(p.value, 2)))
 AOV_Sex %>% flextable() %>%
  set_header_labels(data = "Variable", p.value = "p-value") %>%
  add_header_lines(values = "ANOVA Results with Sex") %>%
  autofit()%>% align(part="header", align="center") 

anova<-bind_rows(
  mutate(tidy(aov(BMI ~ Sex, vao)) %>% filter(term == "Sex") ,pair="BMI and Sex"),
  mutate(tidy(aov(LHOMAIR~Sex,vao)) %>% filter(term == "Sex"),pair="HOMA-IR and Sex"),
  mutate(tidy(aov(Age ~ Sex, vao)) %>% filter(term == "Sex") ,pair="Age and Sex"))%>%
  filter(term!="Residuals") %>%
  mutate(p.value=round(p.value,3)) %>%
  select(pair, p.value)  

#spearman
A<-tidy(cor.test(vao$BMI, vao$Age, method="spearman"))
B<-tidy(cor.test(vao$LHOMAIR, vao$BMI, method="spearman"))
C<-tidy(cor.test(vao$Age, vao$LHOMAIR, method="spearman"))
correlation<-bind_rows(
  mutate(A, pair="BMI and Age"),
  mutate(B, pair="BMI and HOMA-IR"),
  mutate(C, pair="Age and HOMA-IR"))%>%
  mutate(p.value=round(p.value,3), estimate=round(estimate,2))%>%
 select(pair, estimate, p.value) 

spearman_table <- correlation %>%
  flextable() %>%
  set_header_labels(pair = "Variable Pair", estimate = "rho", p.value = "p-value") %>%
  add_header_lines(values = "Spearman Correlation") %>%
  autofit()%>% align(part="header", align="center") %>%
  add_footer_lines(values = "These variables will be used for model building along with the variable Sex") %>%
  fontsize(part = "footer", size = 9) 

anova_table <- anova %>%
  flextable() %>%
  set_header_labels(pair = "Variable Pair", p.value = "p-value") %>%
  add_header_lines(values = "ANOVA Results") %>%
  autofit()%>% align(part="header", align="center") %>%
  add_footer_lines(values = "Sex will be used for model building along with BMI, HOMA-IR, and Age") %>%
  fontsize(part = "footer", size = 9)

BA<-tidy(cor.test(vao$BMI, vao$Adiponectin, method="spearman"))
BL<-tidy(cor.test(vao$BMI, vao$Leptin, method="spearman"))
HA<-tidy(cor.test(vao$HOMAIR, vao$Adiponectin, method="spearman"))
HL<-tidy(cor.test(vao$HOMAIR, vao$Leptin, method="spearman"))
ALcorr<-bind_rows(
  mutate(BA, pair="BMI and Adiponectin"),
  mutate(BL, pair="BMI and Leptin"),
  mutate(HA, pair="HOMA-IR and Adiponectin"),
  mutate(HL, pair="HOMA-IR and Leptin"))%>%
  mutate(estimate=round(estimate,2),
         p.value=ifelse(p.value<0.01, 
                        formatC(p.value, format = "e", digits = 2), round(p.value, 2)))%>%
 select(pair, estimate, p.value) 

ALcorr %>% flextable() %>% set_header_labels(pair = "Variable Pair", estimate = "rho", p.value = "p-value") %>% add_header_lines(values = "Spearman Correlation") %>% autofit()%>% align(part="header", align="center")

anova_table
spearman_table

```

```{r scatterplot matrix, include=FALSE, echo=FALSE, fig.width=6, fig.height=6 }
##all variables
scatterplot_df <- data.frame(vao %>% select(Sex, BMI, HOMAIR,
                                         NAFLDFS, eGFR, Age))
colnames(scatterplot_df) <-c("Sex", "BMI", "HOMA-IR", "NAFLD-FS", "eGFR", "Age")

custom_points <- function(data, mapping, ...) {
  ggplot(data, mapping) +
    geom_point(alpha = 0.5, color = "blue4", size = 1.5, ...)
  }
my_fn<-function(data,mapping,method="spearman",use="pairwise.comparison",...){
  x <- eval_data_col(data, mapping$x)
  y <- eval_data_col(data, mapping$y)
  
  # calculate correlation
  test <- cor.test(x, y, method = method, use = use)
  rho <- round(test$estimate, 2)
  p_value <- if (test$p.value < 0.001) {
    formatC(test$p.value, format = "e", digits = 2)
  } else {
    round(test$p.value, 3)
  }
  
  label <- bquote(atop(rho == .(rho), ~p== .(p_value)))
  grid.text(label, x = 0.5, y = 0.5, gp = gpar(fontsize = 9))
}

ggpairs(scatterplot_df, 
        lower=list(continuous = wrap(custom_points)),
        upper=list(continuous = wrap(my_fn)),
        diag=list(continuous = wrap("densityDiag")),
        title="Scatterplot Matrix of Variables")
```


```{r variable matrix, echo=FALSE, fig.width=7, fig.height=7}
custom_points <- function(data, mapping, ...) {
  ggplot(data, mapping) +
    geom_point(alpha = 0.5, color = "blue4", size = 1.5, ...)
}

my_fn<-function(data,mapping,method="spearman",use="pairwise.comparison",...){
  x <- eval_data_col(data, mapping$x)
  y <- eval_data_col(data, mapping$y)
  test <- cor.test(x, y, method = method, use = use)
  rho <- round(test$estimate, 2)
  p_value <- if (test$p.value < 0.001) {
    formatC(test$p.value, format = "e", digits = 2)
  } else {
    round(test$p.value, 3)
  }
  
  label <- bquote(atop(rho == .(rho), ~p== .(p_value)))
  grid.text(label, x = 0.5, y = 0.5, gp = gpar(fontsize = 9))}

plain_density <- function(data, mapping, ...) {
  ggplot(data, mapping) +
    geom_density(fill = "grey80", color = "black", alpha = 0.7, ...)}

scatterplot_df <- data.frame(vao %>% select(Sex, BMI, HOMAIR,
                                         NAFLDFS, eGFR, Age))

colnames(scatterplot_df)<-c("Sex", "atop(BMI, (kg/m^2))", "HOMA-IR", "NAFLD-FS",
                            "atop(eGFR, (mL/min/1.73~m^2))", "atop(Age, (years))")

p<- ggpairs(scatterplot_df,
          lower = list(continuous = wrap(custom_points)),
          upper = list(continuous = wrap(my_fn)),
          diag = list(continuous = wrap(plain_density)),
          title = "                                   Scatterplot Matrix of Variables",
          mapping = aes(color = Sex), labeller =  label_parsed)

grid.newpage()
grid.draw(GGally::ggmatrix_gtable(p))
grid.rect(x = unit(0, "npc"), y = unit(0, "npc"),
          width = unit(0.38, "npc"), height = unit(3, "npc"),
          gp = gpar(fill = "white", col = NA))

```

\newpage

<!-- For PDF -->

# RBP4 Model

```{r RBP4 model building}
#log transform RBP4 concentrations
vao$LRBP4<-log10(vao$RBP4)

#linear regression with BMI, Sex, Age and HOMAIR
model1=lm(LRBP4~BMI+Sex+Age+LHOMAIR,vao)
vif(model1)

#remove one variable at a time
model2=lm(LRBP4~BMI+Sex+Age,vao) #remove HOMAIR
model3=lm(LRBP4~BMI+Sex+LHOMAIR, vao) #remove Age
model4=lm(LRBP4~BMI+Age+LHOMAIR, vao) #remove Sex
model5=lm(LRBP4~Sex+Age+LHOMAIR, vao) #remove BMI

AIC<-AIC(model1,model2,model3,model4,model5) #summary of AIC
AIC_df<-data.frame(
  Model=c("Model 1: Full Model", "Model 2: No HOMA-IR", "Model 3: No Age",
          "Model 4: No Sex", "Model 5: No BMI"),
  AIC = round(AIC$AIC, 3),
  DF = AIC$df)

aic_table <- AIC_df %>%
  flextable() %>%
  set_header_labels(Model = "Model Description", AIC = "AIC Value",
                    DF="Degrees of Freedom") %>%
  add_header_lines(values = "AIC Comparison of Models") %>%
  align(part="header", align="center" ) %>%
  add_footer_lines(values = "Note: Lower AIC values indicate a better model.") %>%
  fontsize(part = "footer", size = 8) %>%
  set_table_properties(layout = "autofit")
aic_table

##remove BMI (lowest AIC), add interaction terms
model6=lm(LRBP4~Sex+Age+LHOMAIR+Age*Sex+LHOMAIR*Sex,vao)
summary(model6)
AIC(model6,model5)

#remove interaction terms one at a time
model7=lm(LRBP4~Sex+Age+LHOMAIR+LHOMAIR*Age,vao)
model8=lm(LRBP4~Sex+Age+LHOMAIR+Age*Sex,vao)
summary(model7)
summary(model8)
AIC(model6,model7,model8) #remove HOMAIR x Age 

AIC_RBP4_iterm<-AIC(model6,model7,model8) #summary of AIC
AIC_df_RBP4_iterm<-data.frame(
  Model=c("Model 6: Includes interaction terms", "Model 7: Remove Age x Sex", 
          "Model 8: Remove HOMA-IR x Sex"),
  AIC = round(AIC_RBP4_iterm$AIC,3),
  DF = AIC_RBP4_iterm$df)

aic_RBP4_iterm_table <- AIC_df_RBP4_iterm %>%
  flextable() %>%
  set_header_labels(Model = "Model Description", AIC = "AIC Value",
                    DF="Degrees of Freedom") %>%
  add_header_lines(values = "AIC Comparison of Models") %>%
  align(part="header", align="center" ) %>%
  add_footer_lines(values = "Note: Lower AIC values indicate a better model.") %>%
  fontsize(part = "footer", size = 8) %>%
  set_table_properties(layout = "autofit")
aic_RBP4_iterm_table

#remove HOMA-IR
model9=lm(LRBP4~Sex+Age+Age*Sex,vao)
summary(model9)
AIC(model8,model9) # removing HOMA-IR leads to lower AIC

#does adding BMI back into the model improve the model?
model10=lm(LRBP4~Sex+Age+Age*Sex+BMI,vao)
summary(model10)
AIC(model9,model10)
anova(model9,model10) #no, both from ANOVA and AIC, BMI does not improve the model

#does the interaction term improve the model?
model11=lm(LRBP4~Sex+Age,vao)
AIC(model9,model11)
anova(model9,model11) #yes

#is this the simplest model?
model12=lm(LRBP4~Sex,vao) #Sex alone
model13=lm(LRBP4~Age,vao) #Age alone

#summary of AIC
 AICR_3 <- AIC(model9,model12,model13)
 AICR_3df <- data.frame(
   Model = c("Age + Sex + Age x Sex", "Age", "Sex"),
   AIC = round(AICR_3$AIC, 3), 
   DF = AICR_3$df)
 AICR_3df %>%  flextable() %>%
   set_header_labels(Model = "Model Description", AIC = "AIC Value",
                     DF = "Degrees of Freedom") %>%
   add_header_lines(values = "AIC Comparison of Models for RBP4") %>%
   align(part = "header", align = "center") %>%
   add_footer_lines(values = "Note: Lower AIC values indicate a better model.") %>%
fontsize(part = "footer", size = 8) %>%
set_table_properties(layout = "autofit")

#summary of anova results
 p_values<-data.frame(
   Comparison=c("Age+Sex+AgexSex v Age", "Age+Sex+AgexSex v Sex"),
   P_value=c(round(anova(model9,model12)$"Pr(>F)"[2],3),
             round(anova(model9,model13)$"Pr(>F)"[2],3)))
 
 p_values %>% flextable() %>%
     set_header_labels(Comparison="Models", P_value="P-value: Probability >F") %>%
     add_header_lines(values="Is this the simplest model that fits the data best?") %>%
     align(part="header", align="center" ) %>%
     set_table_properties(layout = "autofit") %>%
     add_footer_lines(values = "Note: p<0.05 cut-off for retaining more complex model") %>%
     fontsize(part = "footer", size = 8) 
 
#lower AIC, and anova p<0.05 for the more complex model of Age + Sex + Age x Sex
##final model##
summary(model9)
#variance inflation factors 
vif(model9,type=c("predictor")) #because this model has interaction terms we need to use the GVIF
```

## RBP4 Final Model

> RBP4 Final Model: RBP4 \~ Age + Sex + Age x Sex (p=0.009)

## RBP4 Model Plots

```{r plots for RBP4, echo=FALSE,  fig.width=5, fig.height=5,fig.show="hold", out.width="50%" }
#plot for log rbp4, Age and Sex with separate regression lines segmented by Sex
ggplot(vao,aes(Age,LRBP4))+
  geom_point(aes(color=Sex))+
  scale_color_manual(values = c("Female"="maroon", "Male"="blue4"))+
  geom_smooth(method="lm", aes(group=Sex, color=Sex, fill=Sex), se=TRUE, linewidth=0.5, linetype="dashed")+
  scale_fill_manual(values = c("Female" = "thistle", "Male" = "lightsteelblue2"))+
  labs(title=expression("RBP4 Age and Sex"),
       x=expression("Age (years)"), y=expression("Log RBP4 ("*mu*"M)"))+
  theme_minimal()+
  scale_x_continuous(breaks=seq(25,65,by=5),labels=seq(25,65,by=5),
                     minor_breaks=seq(25,65), limits=c(25,65))+
  scale_y_continuous(breaks=seq(-0.1,.6,by=0.1),labels=seq(-0.1,.6,by=0.1),
                     minor_breaks=seq(-0.1,.6,by=0.05), limits=c(-0.1,.6))+
  coord_fixed(ratio=57)+
  theme(panel.grid.major = element_line(linewidth=0.5, linetyp="dotted", color="grey85"),
        panel.grid.minor=element_line(linewidth=0.5, linetype="dotted"),
        plot.title=element_text(size=12,color="black",hjust=0.5))

#plot for rbp4, Age and Sex with separate regression lines segmented by Sex
ggplot(vao,aes(Age,RBP4))+
  geom_point(aes(color=Sex))+
  scale_color_manual(values = c("Female"="maroon", "Male"="blue4"))+
  geom_smooth(method="lm", aes(group=Sex, color=Sex, fill=Sex), se=TRUE, linewidth=0.5, linetype="dashed")+
  scale_fill_manual(values = c("Female" = "thistle", "Male" = "lightsteelblue2"))+
  labs(title=expression("RBP4 Age and Sex"),
       x=expression("Age (years)"), y=expression("RBP4 ("*mu*"M)"))+
  theme_minimal()+
  scale_x_continuous(breaks=seq(25,65,by=5),labels=seq(25,65,by=5),
                     minor_breaks=seq(25,65), limits=c(25,65))+
  scale_y_continuous(breaks=seq(.75,3.5,by=0.5),labels=seq(.75,3.5,by=0.5),
                     minor_breaks=seq(.75,3.5,by=0.5), limits=c(.75,3.5))+
  coord_fixed(ratio=14.5)+
  theme(panel.grid.major = element_line(linewidth=0.5, linetyp="dotted", color="grey85"),
        panel.grid.minor=element_line(linewidth=0.5, linetype="dotted"),
        plot.title=element_text(size=12,color="black",hjust=0.5))
```

\newpage

<!-- For PDF -->

## Post-hoc analysis of RBP4 and eGFR
```{r post hoc rbp4 and egFR, echo=TRUE}
summary(lm(LRBP4~eGFR,vao))
summary(lm(LRBP4~eGFR*Sex,vao))
```

> eGFR is not a significant correlate for RBP4 (p=0.12)
> eGFR as an interaction term with sex is also not significant (p=0.16) 

\newpage

<!-- For PDF -->

# TTR Model

```{r TTR model building}
vao$LTTR<-log10(vao$TTR)

#linear regression with BMI, Sex, Age and HOMA-IR
Tmodel1=lm(LTTR~BMI+Sex+Age+LHOMAIR,vao)
summary(Tmodel1)
vif(Tmodel1)

#remove one variable at a time
Tmodel2=lm(LTTR~BMI+Sex+Age,vao) #remove HOMA-IR
Tmodel3=lm(LTTR~BMI+Sex+LHOMAIR, vao) #remove Age
Tmodel4=lm(LTTR~BMI+Age+LHOMAIR, vao) #remove Sex
Tmodel5=lm(LTTR~Sex+Age+LHOMAIR, vao) #remove BMI

AIC(Tmodel1,Tmodel2,Tmodel3,Tmodel4,Tmodel5) #summary of AIC

AICT<-AIC(Tmodel1,Tmodel2,Tmodel3,Tmodel4,Tmodel5) #summary of AIC
AICT_df<-data.frame(
  Model=c("Model 1: Full Model", "Model 2: No HOMA-IR", "Model 3: No Age", 
          "Model 4: No Sex", "Model 5: No BMI"),
  AIC = round(AICT$AIC,3), DF = AICT$df)

aicT_table <- AICT_df %>%
  flextable() %>%
  set_header_labels(Model = "Model Description", AIC = "AIC Value",
                    DF="Degrees of Freedom") %>%
  add_header_lines(values = "AIC Comparison of Models for TTR") %>%
  align(part="header", align="center" ) %>%
  add_footer_lines(values = "Note: Lower AIC values indicate a better model.") %>%
  fontsize(part = "footer", size = 8) %>%
  set_table_properties(layout = "autofit")
aicT_table


##remove HOMA-IR (lowest AIC)
#add interaction terms for Age and Sex which are not correlated
Tmodel6=lm(LTTR~Sex+Age+BMI+Age*Sex,vao)
summary(Tmodel6)
AIC(Tmodel6,Tmodel2)
anova(Tmodel6,Tmodel2)
##inclusion of interaction term improves model
###Model now TTR~Sex+BMI+Age+Age*Sex

#remove terms
Tmodel7=lm(LTTR~Sex+Age+Age*Sex,vao) # remove BMI
Tmodel8=lm(LTTR~Sex+BMI,vao) # remove Age
Tmodel9=lm(LTTR~Age+BMI,vao) #remove Sex

AICT_iterm<-AIC(Tmodel6,Tmodel7,Tmodel8,Tmodel9) #summary of AIC
AICT_iterm_df<-data.frame(
  Model=c("Full Model: Sex+Age+BMI+AgexSex",
          "Remove BMI: Sex+Age+AgexSex",
          "Remove Age: Sex+BMI",
          "Remove Sex: Age+BMI"),
  AIC = round(AICT_iterm$AIC,3), DF = AICT_iterm$df)

aicT_iterm_table <- AICT_iterm_df %>%
  flextable() %>%
  set_header_labels(Model = "Model Description", AIC = "AIC Value",
                    DF="Degrees of Freedom") %>%
  add_header_lines(values = "AIC Comparison of Models for TTR") %>%
  align(part="header", align="center" ) %>%
  add_footer_lines(values = "Note: Lower AIC values indicate a better model.") %>%
  fontsize(part = "footer", size = 8) %>%
  set_table_properties(layout = "autofit")
aicT_iterm_table

#lowest AIC removing BMI
#confirm with anova
anova(Tmodel7,Tmodel8)
anova(Tmodel6,Tmodel7) # removing BMI leads to lower AIC, stat. sig.

###Model now TTR ~ Sex + Age + Age * Sex

#does adding HOMA-IR back into the model improve the model?
Tmodel10=lm(LTTR~Sex+Age+Age*Sex+LHOMAIR,vao)
summary(Tmodel10)
AIC(Tmodel7, Tmodel10)
anova(Tmodel7, Tmodel10) #no, both from ANOVA and AIC, HOMA does not improve the model

#does the interaction term improve the model?
Tmodel11=lm(LTTR~Sex+Age,vao)
AIC(Tmodel7,Tmodel11)
anova(Tmodel7,Tmodel11) #yes

#is this the simplest model?
Tmodel12=lm(LTTR~Sex,vao) #Sex alone
Tmodel13=lm(LTTR~Age,vao) #Age alone


AICT2<-AIC(Tmodel7,Tmodel11,Tmodel12,Tmodel13) #summary of AIC
AICT2_df<-data.frame(
  Model=c("Sex + Age + Sex x Age",
          "Sex + Age","Sex", "Age"),
  AIC = round(AICT2$AIC,3), DF = AICT2$df)

aicT2_table <- AICT2_df %>%
  flextable() %>%
  set_header_labels(Model = "Model Description", AIC = "AIC Value",
                    DF="Degrees of Freedom") %>%
  add_header_lines(values = "AIC Comparison of Models for TTR") %>%
  align(part="header", align="center" ) %>%
  add_footer_lines(values="Note: Lower AIC values indicate a better model.") %>%
  fontsize(part = "footer", size = 8) %>%
  set_table_properties(layout = "autofit")
aicT2_table

 #summary of anova results
  p_values<-data.frame(
    Comparison=c("Age+Sex+AgexSex v Age+Sex", "Age+Sex+AgexSex v Age", 
                 "Age+Sex+AgexSex v Sex"),
    P_value=c(round(anova(Tmodel7,Tmodel11)$"Pr(>F)"[2],3),
              round(anova(Tmodel7,Tmodel12)$"Pr(>F)"[2],3),
              round(anova(Tmodel7,Tmodel13)$"Pr(>F)"[2],3)))
  
  p_values %>% flextable() %>%
      set_header_labels(Comparison="Models", P_value="P-value: Probability >F") %>%
      add_header_lines(values="Is this the simplest model that fits the data best?") %>%
      align(part="header", align="center" ) %>%
      set_table_properties(layout = "autofit") %>%
      add_footer_lines(values = "Note: p<0.05 cut-off for retaining more complex model") %>%
      fontsize(part = "footer", size = 8) 

  ##final model for TTR
summary(Tmodel7)
#variance inflation factors 
vif(Tmodel7,type=c("predictor")) #because this model has interaction terms we need to use the GVIF
```

## TTR Final Model

> TTR Final Model: TTR \~ Age + Sex + Age x Sex (p=0.02)

## TTR Model Plots

```{r TTR plots, echo=FALSE, fig.width=5, fig.height=5, fig.show="hold", out.width="50%" }
#plot for log ttr, Age and Sex with separate regression lines segmented by Sex
ggplot(vao,aes(Age,LTTR))+
  geom_point(aes(color=Sex))+
  scale_color_manual(values = c("Female"="maroon", "Male"="blue4"))+
  geom_smooth(method="lm", aes(group=Sex, color=Sex, fill=Sex), se=TRUE, linewidth=0.5, linetype="dashed")+
  scale_fill_manual(values = c("Female" = "thistle", "Male" = "lightsteelblue2"))+
  labs(title=expression("TTR Age and Sex"),
       x=expression("Age (years)"), y=expression("Log TTR ("*mu*"M)"))+
  theme_minimal()+
  scale_x_continuous(breaks=seq(25,65,by=5),labels=seq(25,65,by=5),
                     minor_breaks=seq(25,65), limits=c(25,65))+
  scale_y_continuous(breaks=seq(1,1.5,by=0.1),labels=seq(1,1.5,by=0.1),
                     minor_breaks=seq(1,1.5,by=0.05), limits=c(1,1.5))+
  coord_fixed(ratio=80)+
  theme(panel.grid.major = element_line(linewidth=0.5, linetyp="dotted", color="grey85"),
        panel.grid.minor=element_line(linewidth=0.5, linetype="dotted"),
        plot.title=element_text(size=12,color="black",hjust=0.5))

#plot for TTR, Age and Sex with separate regression lines segmented by Sex
ggplot(vao,aes(Age,TTR))+
  geom_point(aes(color=Sex))+
  scale_color_manual(values = c("Female"="maroon", "Male"="blue4"))+
  geom_smooth(method="lm", aes(group=Sex, color=Sex, fill=Sex), se=TRUE, linewidth=0.5, linetype="dashed")+
  scale_fill_manual(values = c("Female" = "thistle", "Male" = "lightsteelblue2"))+
  labs(title=expression("TTR Age and Sex"),
       x=expression("Age (years)"), y=expression("TTR ("*mu*"M)"))+
  theme_minimal()+
  scale_x_continuous(breaks=seq(25,65,by=5),labels=seq(25,65,by=5),
                     minor_breaks=seq(25,65), limits=c(25,65))+
  scale_y_continuous(breaks=seq(10,35,by=5),labels=seq(10,35,by=5),
                     minor_breaks=seq(10,35,by=1), limits=c(10,35))+
  coord_fixed(ratio=1.6)+
  theme(panel.grid.major = element_line(linewidth=0.5, linetyp="dotted", color="grey85"),
        panel.grid.minor=element_line(linewidth=0.5, linetype="dotted"),
        plot.title=element_text(size=12,color="black",hjust=0.5))
```

\newpage

<!-- For PDF -->

# Retinol Model

```{r retinol model builing}
vao$LROL<-log10(vao$Sretinol)

#linear regression with BMI, Sex, Age and HOMAIR
Rmodel1=lm(LROL~BMI+Sex+Age+LHOMAIR,vao)
vif(Rmodel1)

#remove one variable at a time
Rmodel2=lm(LROL~BMI+Sex+Age,vao) #remove HOMA-IR
Rmodel3=lm(LROL~BMI+Sex+LHOMAIR, vao) #remove Age
Rmodel4=lm(LROL~BMI+Age+LHOMAIR, vao) #remove Sex
Rmodel5=lm(LROL~Sex+Age+LHOMAIR, vao) #remove BMI

AIC(Rmodel1,Rmodel2,Rmodel3,Rmodel4,Rmodel5) #summary of AIC

AICR<-AIC(Rmodel1,Rmodel2,Rmodel3,Rmodel4,Rmodel5) #summary of AIC
AICR_df<-data.frame(
  Model=c("Model 1: Full Model", "Model 2: No HOMA-IR", "Model 3: No Age",
          "Model 4: No Sex", "Model 5: No BMI"),
  AIC = AICR$AIC,
  DF = AICR$df)
aicR_table <- AICR_df %>%
  flextable() %>%
  set_header_labels(Model = "Model Description", AIC = "AIC Value",
                    DF="Degrees of Freedom") %>%
  add_header_lines(values = "AIC Comparison of Models for Retinol") %>%
  align(part="header", align="center" ) %>%
  add_footer_lines(values = "Note: Lower AIC values indicate a better model.") %>%
  fontsize(part = "footer", size = 8) %>%
  set_table_properties(layout = "autofit")
aicR_table

##remove HOMA-IR (lowest AIC), add interaction term
Rmodel6=lm(LROL~Age+BMI+Sex+Age*Sex,vao)
summary(Rmodel6)

#can we justify inclusion of interaction terms?
AIC(Rmodel6,Rmodel2)
anova(Rmodel6,Rmodel2) 
#cannot justify interaction terms

#remove each term
Rmodel7=lm(LROL~Age+BMI,vao) #remove Sex
Rmodel8=lm(LROL~Age+Sex,vao) # remove BMI
Rmodel9=lm(LROL~BMI+Sex,vao) # remove Age

p_values<-data.frame(
  Comparison=c("Age+Sex+BMI v Age+BMI", "Age+Sex+BMI v Age+Sex", 
               "Age+Sex+BMI v BMI+Sex"),
  P_value=c(round(anova(Rmodel2,Rmodel7)$"Pr(>F)"[2],3),
            round(anova(Rmodel2,Rmodel8)$"Pr(>F)"[2],3),
            round(anova(Rmodel2,Rmodel9)$"Pr(>F)"[2],3)))
  
  p_values %>% flextable() %>%
    set_header_labels(Comparison="Comparison", P_value="P-value: Probability >F") %>%
    add_header_lines(values="ANOVA Comparison") %>%
    align(part="header", align="center" ) %>%
    set_table_properties(layout = "autofit") %>%
    add_footer_lines(values = "Note: p<0.05 cut-off for retaining more complex model") %>%
    fontsize(part = "footer", size = 8) 

{AICR_2<-AIC(Rmodel2,Rmodel7,Rmodel8,Rmodel9) #summary of AIC
AICR_2df<-data.frame(
  Model=c("Age + BMI + Sex","Age + BMI","Age + Sex", 
          "BMI + Sex"),
  AIC = AICR_2$AIC, DF = AICR_2$df)

aicR_2_table <- AICR_2df %>%
  flextable() %>%
  set_header_labels(Model = "Model Description", AIC = "AIC Value",
                    DF="Degrees of Freedom") %>%
  add_header_lines(values = "AIC Comparison of Models for Retinol") %>%
  align(part="header", align="center" ) %>%
  add_footer_lines(values = "Note: Lower AIC values indicate a better model.") %>%
  fontsize(part = "footer", size = 8) %>%
  set_table_properties(layout = "autofit")}
aicR_2_table

#Age and Sex has the lowest AIC
#single regressions
Rmodel10=lm(LROL~Age,vao) 
Rmodel11=lm(LROL~Sex,vao)
AIC(Rmodel11, Rmodel10,Rmodel8)
anova(Rmodel10,Rmodel8) #not better than Age alone
anova(Rmodel11,Rmodel8) #not better than Sex alone

summary(Rmodel10)
summary(Rmodel11)

#what about BMI alone
Rmodel12=lm(LROL~BMI,vao)
summary(Rmodel12)
#best p-value for BMI alone

{AICR_3<-AIC(Rmodel7,Rmodel8,Rmodel9,Rmodel10,Rmodel11,Rmodel12) #summary of AIC
AICR_3df<-data.frame(
  Model=c("Age + BMI","Age + Sex", 
          "BMI + Sex", "Age", "Sex", "BMI"),
  AIC = AICR_3$AIC, DF = AICR_3$df)

aicR_3_table <- AICR_3df %>%
  flextable() %>%
  set_header_labels(Model = "Model Description", AIC = "AIC Value",
                    DF="Degrees of Freedom") %>%
  add_header_lines(values = "AIC Comparison of Models for Retinol") %>%
  align(part="header", align="center" ) %>%
  add_footer_lines(values = "Note: Lower AIC values indicate a better model.") %>%
  fontsize(part = "footer", size = 8) %>%
  set_table_properties(layout = "autofit")}
aicR_3_table

#BMI alone also has lowest AIC
##final model is ROL ~ BMI
```

## Retinol Final Model

> Retinol Final Model: Retinol \~ BMI (p=0.016)

## Retinol Model Plots

```{r retinol plots, echo=FALSE,fig.width=4, fig.height=4}
#plot for retinol and BMI
ggplot(vao,aes(BMI,Sretinol))+ geom_point()+
  geom_smooth(method="lm", se=TRUE, fill="wheat", linewidth=0.5, color="goldenrod4")+
  labs(title=expression("Retinol and BMI"),
       x=expression("BMI (kg/m"^2*")"), y=expression("Retinol ("*mu*"M)"))+
  theme_minimal()+
  scale_x_continuous(breaks=seq(20,60,by=5),labels=seq(20,60,by=5),
                     minor_breaks=seq(20,60,by=2.5), limits=c(20,60))+
  scale_y_continuous(breaks=seq(.5,3,by=0.25),labels=seq(.5,3,by=0.25),
                     minor_breaks=seq(.5,3,by=0.25), limits=c(.5,3))+
  coord_fixed(ratio=16)+
  theme(panel.grid.major = element_line(linewidth=0.5, linetype="dotted", color="grey85"),
        panel.grid.minor=element_line(linewidth=0.5, linetype="dotted"),
        plot.title=element_text(size=12, face="bold", color="black",hjust=0.5))
```

```{r extra retinol plots, include=FALSE, echo=FALSE}
#plot for retinol and Sex
ggplot(vao,aes(Sex,Sretinol))+geom_boxplot(outlier.shape=NA)+geom_point()+
  labs(title=expression("Retinol and Sex"),
    x = "Sex",
    y =expression("Retinol ("*mu*"M)")) + theme_minimal() +
  theme(
    plot.title = element_text(size = 12,color = "black",hjust = 0.5)) 

#plot for retinol and Age
ggplot(vao,aes(Age,Sretinol))+ geom_point()+
  geom_smooth(method="lm", se=TRUE, fill="lightsteelblue1", linewidth=0.5, linetype="dashed")+
  labs(title=expression("Retinol and Age"),
       x=expression("Age (years)"), y=expression("Retinol ("*mu*"M)"))+
  theme_minimal()+
scale_x_continuous(breaks=seq(25,65,by=5),labels=seq(25,65,by=5),
                     minor_breaks=seq(25,65), limits=c(25,65))+
  scale_y_continuous(breaks=seq(.5,3,by=0.25),labels=seq(.5,3,by=0.25),
                     minor_breaks=seq(.5,3,by=0.25), limits=c(.5,3))+
  coord_fixed(ratio=16)+
  theme(panel.grid.major = element_line(linewidth=0.5, linetyp="dotted", color="grey85"),
        panel.grid.minor=element_line(linewidth=0.5, linetype="dotted"),
        plot.title=element_text(size=12,color="black",hjust=0.5))
```

\newpage

<!-- For PDF -->

## Sensitivity Analysis of Retinol and BMI
```{r retinol and BMI sensitivity analysis, echo=TRUE}
vaono23 <-vao %>% filter(ID!= "51-3523") #omit 23 from data set
Rmodel12no23<-lm(LROL~BMI,vaono23)
summary(Rmodel12no23)
```

> Omission of participant with liver fibrosis and overt retinol deficiency from the regression model increased the p-value of correlation between BMI and retinol to p=0.09

## Retinol Regressions Omitting Participant with Retinol Deficiency and Apparent Liver Fibrosis
```{r retinol regression omitting one participant, echo=TRUE}
Rno23model1=lm(LROL~BMI+Sex+Age+LHOMAIR,vaono23)
summary(Rno23model1)

#remove one variable at a time
Rno23model2=lm(LROL~BMI+Sex+Age,vaono23) #remove HOMAIR
Rno23model3=lm(LROL~BMI+Sex+LHOMAIR, vaono23) #remove Age
Rno23model4=lm(LROL~BMI+Age+LHOMAIR, vaono23) #remove Sex
Rno23model5=lm(LROL~Sex+Age+LHOMAIR, vaono23) #remove BMI

AICR<-AIC(Rno23model1, Rno23model2, Rno23model3, Rno23model4,Rno23model5) #summary of AIC
AICR_df<-data.frame(
  Model=c("Model 1: Full Model", "Model 2: No HOMA-IR", "Model 3: No Age", 
          "Model 4: No Sex", "Model 5: No BMI"),
  AIC = AICR$AIC,
  DF = AICR$df)
aicR_table <- AICR_df %>%
  flextable() %>%
  set_header_labels(Model = "Model Description", AIC = "AIC Value",
                    DF="Degrees of Freedom") %>%
  add_header_lines(values = "AIC Comparison of Models for Retinol (Omitting 23)") %>%
  align(part="header", align="center" ) %>%
  add_footer_lines(values = "Note: Lower AIC values indicate a better model.") %>%
  fontsize(part = "footer", size = 8) %>%
  set_table_properties(layout = "autofit")
aicR_table

#remove BMI, add interaction terms
Rno23model6=lm(LROL~Sex+Age+LHOMAIR+Sex*Age,vaono23)
Rno23model7=lm(LROL~Sex+Age+Sex*Age,vaono23) #remove HOMA
Rno23model8=lm(LROL~Sex+Age,vaono23) #remove HOMA and Sex * Age
Rno23model9=lm(LROL~Sex+LHOMAIR,vaono23) #remove Age
Rno23model10=lm(LROL~Age+LHOMAIR,vaono23) #remove Sex

AIC(Rno23model5, Rno23model6,Rno23model7,Rno23model8,Rno23model9,Rno23model10)
#model 8 has the lowest AIC
#Sex + Age
summary(Rno23model8)


#how does this compare to the predictors alone
Rno23model11=lm(LROL~Sex,vaono23) #Sex alone
Rno23model12=lm(LROL~Age,vaono23) #Age alone
AIC(Rno23model8, Rno23model11, Rno23model12)
anova(Rno23model8,Rno23model11) #not better than Sex alone
anova(Rno23model8,Rno23model12) #not better than Age alone

summary(Rno23model11) #p>0.05
summary(Rno23model12) #p>0.05
summary(lm(LROL~BMI,vaono23)) 
```

> No significant correlates for retinol after omission participant with gross morphology of liver fibrosis and overt retinol deficiency 

\newpage

<!-- For PDF -->

# Genotype effect on Retinol, RBP4, and TTR
```{r anova for genotypes, echo=FALSE}

anova_table <- data.frame(
  Analyte = c("Retinol", "Retinol", "RBP4", "RBP4", "TTR", "TTR"),
  SNP = c("TTRsnp", "RBP4snp", "TTRsnp", "RBP4snp", "TTRsnp", "RBP4snp"),
  P_value = c(round(summary(aov(Sretinol~TTRsnp,vao))[[1]]$"Pr(>F)"[1],2),
              round(summary(aov(Sretinol~RBP4snp,vao))[[1]]$"Pr(>F)"[1],2),
              round(summary(aov(RBP4~TTRsnp,vao))[[1]]$"Pr(>F)"[1],2),
              round(summary(aov(RBP4~RBP4snp,vao))[[1]]$"Pr(>F)"[1],2),
              round(summary(aov(TTR~RBP4snp,vao))[[1]]$"Pr(>F)"[1],2),
              round(summary(aov(TTR~TTRsnp,vao))[[1]]$"Pr(>F)"[1],2)))
anova_table %>% flextable() %>%
  set_header_labels(
    Analyte = "Analyte",
    SNP = "SNP",
    P_value = "p-value"
  ) %>%
  add_header_lines(values = "ANOVA Results Table") %>%
  align(part = "header", align = "center") %>%
  set_table_properties(layout = "autofit")
```

# Table S2, S3
```{r table s2 and s3, echo=FALSE}
#function to calculate geometric mean
geo_mean<-function(x) {exp(mean(log(x[x>0])))}

summary_RBP4snp<- vao %>% group_by(RBP4snp) %>% summarise(    
  RBP4 = sprintf("%.2f (%.2f, %.2f)", geo_mean(RBP4), min(RBP4), max(RBP4)),
    TTR = sprintf("%.2f (%.2f, %.2f)", geo_mean(TTR), min(TTR), max(TTR)),
    Retinol = sprintf("%.2f (%.2f, %.2f)", geo_mean(Sretinol), min(Sretinol), max(Sretinol)),
    .groups = "drop")

summary_TTRsnp<- vao %>% group_by(TTRsnp) %>% summarise(    
  RBP4 = sprintf("%.2f (%.2f, %.2f)", geo_mean(RBP4), min(RBP4), max(RBP4)),
    TTR = sprintf("%.2f (%.2f, %.2f)", geo_mean(TTR), min(TTR), max(TTR)),
    Retinol = sprintf("%.2f (%.2f, %.2f)", geo_mean(Sretinol), min(Sretinol), max(Sretinol)),
    .groups = "drop")

flextable(summary_RBP4snp) %>% colformat_double(digits=2) %>% 
  set_header_labels(RBP4snp = "Genotype", RBP4="RBP4 (\u03bcM)", TTR = "TTR (\u03bcM)", 
                    Retinol = "Retinol (\u03bcM)") %>% autofit() %>% 
  add_header_lines(as_paragraph("Analyte Conc. Stratified by ",as_i("RBP4")," Genotype (rs10882272)")) %>% 
    add_footer_lines(values = c("", "")) %>%  
  compose(part = "footer", i = 1, j = 1,
    value = as_paragraph("Concentrations shown as geometric mean and range")) %>%
  compose(part = "footer", i = 2, j = 1,
    value = as_paragraph("For the ", as_i("RBP4"), " genotype, 1/1 refers to T/T, 1/2 to T/C and 2/2 to C/C for single nucleotide polymorphism in the 3â€™ untranslated region"))   %>% 
  align(part = "header", align = "center") %>% fontsize(part="footer", size=9)

flextable(summary_TTRsnp) %>% colformat_double(digits=2) %>% 
  set_header_labels(TTRsnp = "Genotype", RBP4="RBP4 (\u03bcM)", TTR = "TTR (\u03bcM)", 
                    Retinol = "Retinol (\u03bcM)") %>% autofit() %>% 
  add_header_lines(as_paragraph("Analyte Conc. Stratified by ",as_i("TTR")," Genotype (rs1667255)")) %>% 
  add_footer_lines(values = c("", "")) %>%  
  compose(part = "footer", i = 1, j = 1,
    value = as_paragraph("Concentrations shown as geometric mean and range")) %>%
  compose(part = "footer", i = 2, j = 1,
    value = as_paragraph("For the ", as_i("TTR"), " genotype, 1/1 refers to A/A, 1/2 to A/C and 2/2 to C/C for intronic single nucleotide polymorphism")) %>% 
  align(part = "header", align = "center") %>% fontsize(part="footer", size=9)
```


\newpage

<!-- For PDF -->

# Figure 2
```{r figure 2, echo=FALSE, fig.width=7, fig.height=2.5}
RBP4_plot<- ggplot(vao,aes(Age,RBP4))+
  geom_point(aes(color=Sex))+
  scale_color_manual(values = c("Female"="maroon", "Male"="blue4"))+
  geom_smooth(method="lm", aes(group=Sex, color=Sex, fill=Sex), se=TRUE, linewidth=0.5, linetype="dashed")+
  scale_fill_manual(values = c("Female" = "thistle", "Male" = "lightsteelblue2"))+
  labs(title=expression(), color="Sex", fill="Sex",
       x=expression("Age (years old)"), y=expression("RBP4 ("*mu*"M)"))+
  theme_minimal()+
  scale_x_continuous(breaks=seq(25,65,by=10),labels=seq(25,65,by=10),
                     minor_breaks=seq(25,65, by=5), limits=c(25,65))+
  scale_y_continuous(breaks=seq(0,3.5,by=0.5),labels=seq(0,3.5,by=0.5),
                     minor_breaks=seq(0,3.5,by=0.5), limits=c(0,3.5))+
  coord_fixed(ratio=11.43)+
  theme(legend.position=c(0.6,0.14), 
        legend.background=element_rect(fill="white",linetype="solid",color="white"),
               legend.text=element_text(size=8), legend.title=element_blank(), 
        legend.direction="horizontal", legend.key.spacing.x=unit(2, "pt"),
        panel.grid.major = element_line(linewidth=0.5, linetype="dotted", color="grey85"),
        panel.grid.minor=element_line(linewidth=0.5, linetype="dotted"),
        axis.line=element_line(size = .5, colour = "black"))

vao23<-data.frame(BMI=c(55.7), Sretinol=c(0.637))

ROL_plot <-ggplot(vao,aes(BMI,Sretinol))+ geom_point()+
  geom_smooth(method="lm", se=TRUE, fill="wheat", linewidth=0.5, color="goldenrod4")+
  geom_point(data = vao23, aes(BMI, Sretinol), color = "red3", shape=17, size=2) +
  labs(title=expression(),
       x=expression("BMI (kg/m"^2*")"), y=expression("Retinol ("*mu*"M)"))+
  theme_minimal()+
  scale_x_continuous(breaks=seq(20,60,by=10),labels=seq(20,60,by=10),
                     minor_breaks=seq(20,60,by=5), limits=c(20,60))+
  scale_y_continuous(breaks=seq(0,3,by=0.5),labels=seq(0,3,by=0.5),
                     minor_breaks=seq(0,3,by=0.5), limits=c(0,3))+
  coord_fixed(ratio=13.33)+
  theme(panel.grid.major = element_line(linewidth=0.5, linetype="dotted", color="grey85"),
        panel.grid.minor=element_line(linewidth=0.5, linetype="dotted"),
        plot.title=element_text(size=12,color="black",hjust=0.5),
  axis.line = element_line(size = .5, colour = "black"))

TTR_plot<- ggplot(vao,aes(Age,TTR))+
  geom_point(aes(color=Sex))+
  scale_color_manual(values = c("Female"="maroon", "Male"="blue4"))+
  geom_smooth(method="lm", aes(group=Sex, color=Sex, fill=Sex), se=TRUE, linewidth=0.5, linetype="dashed")+
  scale_fill_manual(values = c("Female" = "thistle", "Male" = "lightsteelblue2"))+
  labs(x=expression("Age (years old)"), y=expression("TTR ("*mu*"M)"))+
  theme_minimal()+
  scale_x_continuous(breaks=seq(25,65,by=10),labels=seq(25,65,by=10),
                     minor_breaks=seq(25,65), limits=c(25,65))+
  scale_y_continuous(breaks=seq(0,35,by=10),labels=seq(0,35,by=10),
                     minor_breaks=seq(0,35,by=5), limits=c(0,35))+
  coord_fixed(ratio=1.14)+
  theme(legend.position=c(0.6,0.14), 
        legend.background=element_rect(fill="white",linetype="solid",color="white"),
               legend.text=element_text(size=8), legend.title=element_blank(), 
        legend.direction="horizontal", legend.key.spacing.x=unit(2, "pt"),
        panel.grid.major = element_line(linewidth=0.5, linetype="dotted", color="grey85"),
        panel.grid.minor=element_line(linewidth=0.5, linetype="dotted"),
        axis.line=element_line(size = .5, colour = "black"))


RBP4_p<-ggplot()+ annotate("text", x = 1, y = 1, label = "p = 0.009", color = "black", fontface = 'italic') +
  theme_void()
ROL_p<-ggplot()+ annotate("text", x = 1, y = 1, label = "p = 0.016", color = "black",fontface = 'italic') +
  theme_void()
TTR_p<-ggplot()+ annotate("text", x = 1, y = 1, label = "p = 0.022", color = "black", fontface = 'italic') +
  theme_void()


RBP4_comb<-ggarrange(RBP4_p, RBP4_plot, ncol = 1, heights = c(0.07, 1))
ROL_comb<-ggarrange(ROL_p, ROL_plot, ncol = 1, heights = c(0.07, 1))
TTR_comb<-ggarrange(TTR_p, TTR_plot, ncol = 1, heights = c(0.07, 1))


ggarrange(RBP4_comb, TTR_comb, ROL_comb, ncol=3, nrow=1,
                   labels=c("A)", "B)", "C)"),
                   label.x=0,
                   label.y=1)
```
